pkgs/by-name/ri/rimsort/package.nix

```
@@ -0,0 +1,157 @@
+{
+  lib,
+  stdenv,
+  python3Packages,
+  fetchFromGitHub,
+  fetchzip,
+  makeWrapper,
+
+  makeDesktopItem,
+  replaceVars,
+
+  todds,
+
+  steam,
+}:
+let
+  pname = "rimsort";
+  version = "1.0.20";
+
+  srcs = [
+    (fetchFromGitHub {
+      owner = "RimSort";
+      repo = "RimSort";
+      tag = "v${version}";
+      hash = "sha256-ISor56g3zj7ktSmgh+JS0IbSmgrj6Dv4mTPa18F1UgI=";
+      fetchSubmodules = true;
+    })
+    (fetchzip {
+      url = "https://web.archive.org/web/20250527013243/https://partner.steamgames.com/downloads/steamworks_sdk_162.zip"; # Steam sometimes requires auth to download.
+      hash = "sha256-yDA92nGj3AKTNI4vnoLaa+7mDqupQv0E4YKRRUWqyZw=";
+    })
+  ];
+
+  steamfiles = python3Packages.buildPythonPackage {
+    pname = "steamfiles";
+    inherit version;
+    src = "${builtins.elemAt srcs 0}/submodules/steamfiles";
+    dependencies = with python3Packages; [
+      protobuf
+      protobuf3-to-dict
+    ];
+  };
+
+  steam-run =
+    (steam.override {
+      privateTmp = false;
+    }).run;
+in
+
+stdenv.mkDerivation {
+  inherit pname;
+  inherit version;
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    cp -r ${builtins.elemAt srcs 0} source
+    chmod -R 755 source
+    cp ${builtins.elemAt srcs 1}/redistributable_bin/linux64/libsteam_api.so source/
+
+    runHook postUnpack
+  '';
+
+  sourceRoot = "source";
+
+  patches = [
+    (replaceVars ./todds-path.patch { inherit todds; })
+    (replaceVars ./steam-run.patch { inherit steam-run; })
+    ./hide-update-check-dialog.patch
+  ];
+
+  nativeBuildInputs = [
+    makeWrapper
+  ];
+
+  buildInputs =
+    [
+      todds
+      steamfiles
+    ]
+    ++ builtins.attrValues {
+      inherit (python3Packages)
+        beautifulsoup4
+        certifi
+        chardet
+        imageio
+        loguru
+        lxml
+        msgspec
+        natsort
+        networkx
+        platformdirs
+        psutil
+        pygit2
+        pygithub
+        pyperclip
+        pyside6
+        requests
+        sqlalchemy
+        steam
+        toposort
+        watchdog
+        xmltodict
+        steamworkspy
+        ;
+    };
+
+  dontBuild = true;
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "RimSort";
+      desktopName = "RimSort";
+      exec = "rimsort";
+      icon = "io.github.rimsort.rimsort";
+      comment = "RimWorld Mod Manager";
+      categories = [ "Game" ];
+    })
+  ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/usr/lib/rimsort
+    cp -r ./* $out/usr/lib/rimsort/
```

```suggestion
    mkdir -p $out/lib/rimsort
    cp -r ./* $out/lib/rimsort/
```
NixOS does not really have "usr"